File: 53.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        n = len(nums)
        res = 0
        for i in range(n):
            sub = 0
            for j in range(i, n):
                sub += nums[j]
                res = max(res, sub)
        
        return res

File: 53.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int cur = 0, res = 0;
        
        for(int i : nums){
            cur = max(i, cur + i);
            res = max(res, cur);
        }
        return res;
    }
};